backend/src/DataSources/users-data-source.js

```diff
@@ -0,0 +1,65 @@
+const {DataSource} = require('apollo-datasource');
+const {createAccessToken} = require('../token')
+const crypto = require('crypto');
+const bcrypt = require('bcrypt');
+
+class User{
+  constructor(data){
+    this.id = crypto.randomBytes(16).toString('hex')
+    data.password = bcrypt.hashSync(data.password, 10);
```

Comment:
```suggestion
    data.passwordHash = bcrypt.hashSync(data.password, 10);
```
`encryptedPassword` would do as well.